### DISCUSSION TITLE
Relationship panels move to the end of the detail page when using other panels as well

### DISCUSSION THREAD

**Discussion by juhotaipale** [laravel/nova-issues#4179] [Ideas & Feature Requests]
- Laravel Version: 9.11.0
- Nova Version: 4.3.4
- PHP Version: 8.1.5
- Browser type and version: Google Chrome 101.0.4951.54

### Description:
Resource's relationship panels are shown at the bottom of detail page no matter where those are placed in the code.

### Detailed steps to reproduce the issue on a fresh Nova installation:
```php
public function fields(NovaRequest $request)
{
    return [
        ID::make(),
        Text::make('Name'),

        (new Panel('The first panel', [
            Text::make('Some information'),
            Text::make('More information'),
        ]),

        BelongsToMany('The relation related to the first panel'),

        (new Panel('The second panel', [
            Text::make('Some other information'),
            Text::make('More other information'),
        ]),
    ]
}
```

**Comment by crynobone** [comment]
This is the expected behavior right now. 

1. Default panel containing all un-grouped fields (sort fields top to bottom).
2. Custom Panels (sort panels top to bottom).
3. Relationship (sort relationships top to bottom).

**Comment by ianrobertsFF** [comment]
Why? Generally if we put fields in a specific order, they're in that order for UX reasons. Having the framework override that is unhelpful, especially as its an undocumented change that directly contradicts previous behaviour.

Being able to set this behaviour on a per-resource basis would be very useful. Is there a function we can override on the resource to achieve this or is it deeper?

Edit: Looks like overriding `resolvePanelsFromFields` from the `ResolvesFields` trait should fix it in the short term, but this for sure needs some sort of flag on the resource to toggle this behaviour.

**Comment by jbeales** [comment]
I just ran into this with a Resource Tool, which was stuck above my relationships. The resource tool is used much less frequently than the relationships, so I'd like it at the bottom of the page, but Nova is re-arranging my UX here.

Could we turn this into a feature request to allow custom sorting for panels, or simply respect the order that we set them in the Resource's `fields()` method?

Filtering fields without a panel into the top Detail panel makes sense, but the rest is confusing, especially as it's undocumented, and as @ianrobertsFF says, is the opposite of what Nova 1 - 3 did.
